integreat_cms/deepl_api/utils.py

```diff
@@ -58,84 +112,117 @@ def deepl_translation(self, request, content_objects, language_slug, form_class)
+                    messages.error(
+                        request,
+                        _('No source translation could be found for {} "{}".').format(
+                            type(content_object)._meta.verbose_name.title(),
+                            content_object.best_translation.title,
+                        ),
+                    )
+                    continue
+
+                existing_target_translation = content_object.get_translation(
+                    target_language.slug
                 )
-                messages.error(
-                    request,
-                    _('{} "{}" could not be automatically translated.').format(
-                        type(content_object)._meta.verbose_name.title(),
-                        source_translation.title,
-                    ),
+                # For some languages, the DeepL client expects the BCP tag instead of the short language code
+                if target_language.slug in ("en", "pt"):
+                    target_language_key = target_language.bcp47_tag
+                else:
+                    target_language_key = target_language.slug
+
+                # Before translating, check if translation would exceed usage limit
+                (
+                    translation_exceeds_limit,
+                    translation_budget,
+                    translation_word_count,
+                ) = self.check_usage(region, source_translation)
+                if translation_exceeds_limit:
+                    messages.error(
+                        request,
+                        _(
+                            "Translation from {} to {} not possible: translation of {} words would exceed the remaining budget of {} words."
+                        ).format(
+                            source_language,
+                            target_language,
+                            translation_word_count,
+                            max(0, translation_budget - region.deepl_budget_used),
+                        ),
+                    )
+                    continue
+
+                data = {
+                    "status": existing_target_translation.status
+                    if existing_target_translation
+                    else source_translation.status,
+                }
+
+                for attr in self.translatable_attributes:
+                    # Only translate existing, non-empty attributes
+                    if hasattr(source_translation, attr) and getattr(
+                        source_translation, attr
+                    ):
+                        # data has to be unescaped for DeepL to recognize Umlaute
+                        data[attr] = self.translator.translate_text(
+                            unescape(getattr(source_translation, attr)),
+                            source_lang=source_language.slug,
+                            target_lang=target_language_key,
+                            tag_handling="html",
```

Comment:
Since only the `content` field is HTML and `title` and `meta_description` are plain text, I'm not sure whether we need to differentiate here...
But probably there should be no big problems with enabling this option for non-html content, right?